src/core/Clock.js

```diff
@@ -138,8 +142,88 @@ getJasmineRequireObj().Clock = function() {
       }
     };
 
+    /**
+     * Enables or disables automatic advancing of the Clock's timers.
+
+     * When enabled, the time is automatically advanced asynchronously after a macrotask, giving
+     * microtasks and events outside the clock a chance to run before each timer runs.
+     *
+     * @param {boolean} shouldAutoTick
+     */
+    this.setAutoTickMode = function(shouldAutoTick) {
+      const { mode, counter } = autoTickMode;
+      if (shouldAutoTick === mode) {
+        return;
+      }
+      autoTickMode = { mode: shouldAutoTick, counter: counter + 1 };
+      if (shouldAutoTick) {
+        advanceUntilModeChanges();
+      }
+    };
+
     return this;
 
+    /**
+     * Advances the Clock's time until the mode changes.
+     *
+     * The time is advanced asynchronously, giving microtasks and events a chance
+     * to run before each timer runs.
+     *
+     * @function
+     * @return {!Promise<undefined>}
+     */
+    async function advanceUntilModeChanges() {
+      if (!installed) {
+        throw new Error(
+          'Mock clock is not installed, use jasmine.clock().install()'
+        );
+      }
+      const { counter } = autoTickMode;
+
+      while (true) {
+        await newMacrotask();
+
+        if (
+          autoTickMode.counter !== counter ||
+          !installed ||
+          delayedFunctionScheduler === null
+        ) {
+          return;
+        }
+
+        if (!delayedFunctionScheduler.isEmpty()) {
+          delayedFunctionScheduler.runNextQueuedFunction(function(millis) {
+            mockDate.tick(millis);
+          });
+        } else {
+          await Promise.resolve();
```

Comment:
Just wondering about this: is it necessary?

After this line, the very next thing that will happen is going back to the top of the while loop and awaiting newMacrotask. So this could be redundant.

Reply:
Agreed, I don't think it is. Removed.